src/match_result_utils.py

```
@@ -229,16 +229,20 @@ async def fetch_teams(session, match: Match):
-        t2_stmt = select(Team).where(Team.pandascore_id == match.team2_id)
-    else:
-        t2_stmt = select(Team).where(Team.name == match.team2)
+    t1_stmt = _build_team_lookup_stmt(match, match.team1_id, match.team1)
+    t2_stmt = _build_team_lookup_stmt(match, match.team2_id, match.team2)
 
     team1 = (await session.exec(t1_stmt)).first()
     team2 = (await session.exec(t2_stmt)).first()
     return team1, team2
+
+
+def _build_team_lookup_stmt(match: Match, team_id, team_name):
```

Ruff (0.15.5)

[warning] Missing return type annotation for private function `_build_team_lookup_stmt`

(ANN202)